### PR TITLE
Use tailwind

### LINKED ISSUE #2194: Migrate to TailwindCSS
### Discussed in https://github.com/woodpecker-ci/woodpecker/discussions/2147

<div type='discussions-op-text'>

<sup>Originally posted by **6543** July 28, 2023</sup>
https://windicss.org/posts/sunsetting.html

we have to move to a maintained CSS framework and I would propose to use [TailwindCSS](https://github.com/tailwindlabs/tailwindcss) as WindiCSS originated from it

### PR DESCRIPTION
**Pull request by lukashass**
Closes #2194 

## TODO

- [ ] UI fixes
- [ ] rewrite dev docs
- [ ] `hover-effect` alias

### CODE REVIEW

File: web/tailwind.config.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
```
Comment (anbraten) [inline web/tailwind.config.js:1]: tailwind has ts support for its config as well

File: web/package.json [hunk 2]
```diff
@@ -54,19 +56,19 @@
     "eslint-plugin-simple-import-sort": "^10.0.0",
     "eslint-plugin-vue": "^9.20.1",
     "eslint-plugin-vue-scoped-css": "^2.7.2",
+    "postcss": "^8.4.33",
     "prettier": "^3.2.4",
     "replace-in-file": "^7.1.0",
+    "tailwindcss": "^3.4.1",
```
Comment (lafriks) [inline web/package.json:62]: tailwindcss should be moved to dependencies

Comment (silverwind) [reply web/package.json:62]: postcss and autoprefixer too, assuming you keep build dependencies in `dependencies`.

File: web/tailwind.config.ts [hunk 3]
```diff
@@ -24,9 +24,18 @@ const customColors = {
...
   darkMode: 'class',
   theme: {
+    borderWidth: {
+      DEFAULT: '1px',
+      '0': '0',
+      '2': '2px',
+      '4': '4px',
+      '6': '6px',
+      '8': '8px',
+    },
```
Comment (silverwind) [inline web/tailwind.config.ts:38]: Why is this needed?

**Comment by pat-s** [comment]
superseeded by #4614
